src/providers/postgres/qgspostgresconn.h

```diff
@@ -244,15 +244,17 @@ class QgsPostgresConn : public QObject
     // libpq wrapper
     //
 
-    // run a query and check for errors
+    // run a query and check for errors, thread-safe
     PGresult *PQexec( const QString &query, bool logError = true ) const;
     void PQfinish();
     QString PQerrorMessage() const;
-    int PQsendQuery( const QString &query );
     int PQstatus() const;
-    PGresult *PQgetResult();
     PGresult *PQprepare( const QString &stmtName, const QString &query, int nParams, const Oid *paramTypes );
     PGresult *PQexecPrepared( const QString &stmtName, const QStringList &params );
+    //! PQsendQuery and PQgetResult are used for asynchronous queries
+    //! thread safety must be ensure by the caller
```

Comment:
I think there's just a single way to accomplish that, so it would be nice to name it. Can you also copy that to PGgetResult?

Reply:
Sure. Like this ? https://github.com/qgis/QGIS/pull/8700/files#diff-f8272553aa2913d03097e2d92290d134R255

Reply:
Perfect